snuba/utils/streams/abstract.py

```diff
@@ -150,18 +168,26 @@ def tell(self) -> Mapping[TStream, TOffset]:
     @abstractmethod
     def seek(self, offsets: Mapping[TStream, TOffset]) -> None:
         """
-        Change the read offsets for the provided streams.
+        Change the working offsets for the provided streams.
+
+        Raises a ``RuntimeError`` if called on a closed consumer.
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def stage_offsets(self, offsets: Mapping[TStream, TOffset]) -> None:
+        """
+        Stage offsets for the next commit.
 
         Raises a ``RuntimeError`` if called on a closed consumer.
         """
         raise NotImplementedError
 
     @abstractmethod
-    def commit(self) -> Mapping[TStream, TOffset]:
+    def commit_offsets(self) -> Mapping[TStream, TOffset]:
         """
-        Commit staged offsets for all streams that this consumer is assigned
-        to. The return value of this method is a mapping of streams with
-        their committed offsets as values.
```

Comment:
This comment changed, since rather that committing the staged offsets for all streams, it now only commits offsets that were explicitly staged.